src/BinSkim.Driver/RoslynAnalyzer/ILOperations.cs

```diff
@@ -0,0 +1,485 @@
+        public override OperationKind Kind => OperationKind.MethodReferenceExpression;
+    }
+
+    internal abstract class HasOperatorExpression : Expression, IHasOperatorExpression
+    {
+        // operator method calls will be raised as regular invocations so this is always false/null.
+        bool IHasOperatorExpression.UsesOperatorMethod => false;
+        IMethodSymbol IHasOperatorExpression.Operator => null;
+    }
+
+    internal sealed class UnaryOperatorExpression : HasOperatorExpression, IUnaryOperatorExpression
+    {
+        public UnaryOperatorExpression(UnaryOperationKind unaryKind, IExpression operand, ITypeSymbol resultType)
+        {
+            UnaryKind = unaryKind;
+            Operand = operand;
+            ResultType = resultType;
+        }
+
+        public UnaryOperationKind UnaryKind { get; }
+        public IExpression Operand { get; }
+        public override ITypeSymbol ResultType { get; }
+
+        public override OperationKind Kind => OperationKind.UnaryOperatorExpression;
+    }
+
+    internal sealed class BinaryOperatorExpression : HasOperatorExpression, IBinaryOperatorExpression
+    {
+        public BinaryOperatorExpression(BinaryOperationKind binaryKind, IExpression left, IExpression right, ITypeSymbol resultType)
+        {
+            BinaryKind = binaryKind;
+            Left = left;
+            Right = right;
+            ResultType = resultType;
+        }
+
+        public BinaryOperationKind BinaryKind { get; }
+        public IExpression Left { get; }
+        public IExpression Right { get; }
+        public override ITypeSymbol ResultType { get; }
+        public override OperationKind Kind => OperationKind.BinaryOperatorExpression;
+    }
+
+    internal sealed class ConversionExpression : HasOperatorExpression, IConversionExpression
+    {
+        public ConversionExpression(ConversionKind conversion, IExpression operand, ITypeSymbol resultType)
+        {
+            Conversion = conversion;
+            Operand = operand;
+        }
+
+        public ConversionKind Conversion { get; }
+        public IExpression Operand { get; }
+        public override ITypeSymbol ResultType { get; }
+        public override OperationKind Kind => OperationKind.ConversionExpression;
+
+        bool IConversionExpression.IsExplicit => true;
+    }
+
+
+    internal sealed class SizeOfExpression : Expression, ITypeOperationExpression
+    {
+        public SizeOfExpression(ISymbolProvider provider, ITypeSymbol typeOperand)
+        {
+            TypeOperand = typeOperand;
+            ResultType = provider.GetSpecialType(SpecialType.System_UInt32);
+        }
+
+        public ITypeSymbol TypeOperand { get; }
+        public override ITypeSymbol ResultType { get; }
+        public override OperationKind Kind => OperationKind.TypeOperationExpression;
+        public TypeOperationKind TypeOperationClass => TypeOperationKind.SizeOf; // FEEDBACK: Inconsistent naming of enum and member.
```

Comment:
I think @genlu has a PR that reconciles some naming inconsistencies, including this one.